src/System.Management.Automation/engine/parser/Parser.cs

```diff
@@ -4524,72 +4524,122 @@ private StatementAst EnumDefinitionRule(List<AttributeBaseAst> customAttributes,
         {
             //G  enum-statement:
             //G      'enum'   new-lines:opt   enum-name   '{'   enum-member-list   '}'
+            //G      'enum'   new-lines:opt   enum-name   ':'  enum-underlying-type  '{'   enum-member-list   '}'
             //G
             //G  enum-name:
             //G      simple-name
             //G
+            //G  enum-underlying-type:
+            //G      new-lines:opt   valid-type-name   new-lines:opt
+            //G
             //G  enum-member-list:
             //G      enum-member  new-lines:opt
             //G      enum-member-list   enum-member
 
+            const TypeCode validUnderlyingTypeCodes = TypeCode.Byte | TypeCode.Int16 | TypeCode.Int32 | TypeCode.Int64 | TypeCode.SByte | TypeCode.UInt16 | TypeCode.UInt32 | TypeCode.UInt64;
```

Comment:
Const should be upper case:
```suggestion
            const TypeCode ValidUnderlyingTypeCodes = TypeCode.Byte | TypeCode.Int16 | TypeCode.Int32 | TypeCode.Int64 | TypeCode.SByte | TypeCode.UInt16 | TypeCode.UInt32 | TypeCode.UInt64;
```